Game/Source/GDKTestGyms/Public/TestGymsReplicationGraph.h

```diff
@@ -0,0 +1,155 @@
+// Copyright 1998-2019 Epic Games, Inc. All Rights Reserved.
+
+#pragma once
+
+#include "CoreMinimal.h"
+
+#include "EngineClasses/SpatialReplicationGraph.h"
+
+#include "TestGymsReplicationGraph.generated.h"
+
+
+class UReplicationGraphNode_GridSpatialization2D;
+class AGameplayDebuggerCategoryReplicator;
+
+DECLARE_LOG_CATEGORY_EXTERN( LogTestGymsReplicationGraph, Log, All );
+
+// This is the main enum we use to route actors to the right replication node. Each class maps to one enum.
+UENUM()
+enum class EClassRepNodeMapping : uint32
+{
+	NotRouted,						// Doesn't map to any node. Used for special case actors that handled by special case nodes (UTestGymsReplicationGraphNode_PlayerStateFrequencyLimiter)
+	RelevantAllConnections,			// Routes to an AlwaysRelevantNode or AlwaysRelevantStreamingLevelNode node
+	
+	// ONLY SPATIALIZED Enums below here! See UTestGymsReplicationGraph::IsSpatialized
+
+	Spatialize_Static,				// Routes to GridNode: these actors don't move and don't need to be updated every frame.
+	Spatialize_Dynamic,				// Routes to GridNode: these actors mode frequently and are updated once per frame.
+	Spatialize_Dormancy,			// Routes to GridNode: While dormant we treat as static. When flushed/not dormant dynamic. Note this is for things that "move while not dormant".
+};
+
+/** TestGyms Replication Graph implementation. Based on UShooterReplicationGraph */
```

Comment:
@mironec @improbable-danny Like it says, this is heavily based on the UShooterReplicationGraph